### PR TITLE
fallback for date formating in android firefox

### PR DESCRIPTION
**Pull request by jacekkarczmarczyk**
Fix (fallback) for #1655

Date picker in firefox android:
image

With farsi locale:
image

### CODE REVIEW

File: src/components/VDatePicker/VDatePicker.vue [hunk 1]
```diff
@@ -13,6 +13,7 @@
   import Touch from '../../directives/touch'
 
   const defaultDateFormat = val => new Date(val).toISOString().substr(0, 10)
+  const supportsLocaleFormat = ('toLocaleString' in Date.prototype) && new Date(2000, 0, 15).toLocaleDateString('en', { day: 'numeric' }) == '15'
```
Comment (johnleider) [inline src/components/VDatePicker/VDatePicker.vue]: break this at && so it is not too long

File: src/components/VDatePicker/VDatePicker.vue [hunk 2]
```diff
@@ -180,8 +182,8 @@
       const date = new Date()
       date.setDate(date.getDate() - date.getDay() + parseInt(this.firstDayOfWeek))
 
-      createRange(7).forEach(() => {
-        const narrow = date.toLocaleString(this.locale, { weekday: 'narrow' })
+      createRange(7).forEach(index => {
+        const narrow = this.supportsLocaleFormat ? date.toLocaleDateString(this.locale, { weekday: 'narrow' }) : ['S', 'M', 'T', 'W', 'T', 'F', 'S'][(index + parseInt(this.firstDayOfWeek)) % 7]
```
Comment (johnleider) [inline src/components/VDatePicker/VDatePicker.vue]: break this into new lines

File: src/components/VDatePicker/mixins/date-table.js [hunk 3]
```diff
@@ -77,7 +78,7 @@ export default {
               type: 'button'
             },
             domProps: {
-              innerHTML: '<span class="btn__content">' + date.toLocaleDateString(this.locale, { day: 'numeric' }) + '</span>'
```
Comment (johnleider) [inline src/components/VDatePicker/mixins/date-table.js]: could this not remain a template string?

File: src/components/VDatePicker/mixins/date-title.js [hunk 4]
```diff
@@ -50,7 +58,7 @@ export default {
             }
           }
         }, [
-          new Date(this.year, this.month, this.day, 12).toLocaleDateString(this.locale, { year: 'numeric' }),
+          this.supportsLocaleFormat ? new Date(this.year, this.month, this.day, 12).toLocaleDateString(this.locale, { year: 'numeric' }) : this.year,
```
Comment (johnleider) [inline src/components/VDatePicker/mixins/date-title.js]: break this into new lines

File: src/components/VDatePicker/mixins/date-years.js [hunk 5]
```diff
@@ -9,6 +9,7 @@ export default {
     genYearItems () {
       const children = []
       for (let i = this.year + 100, length = this.year - 100; i > length; i--) {
+        const buttonText = this.supportsLocaleFormat ? new Date(i, this.month, this.day, 12).toLocaleDateString(this.locale, { year: 'numeric' }) : i
```
Comment (johnleider) [inline src/components/VDatePicker/mixins/date-years.js]: break this into new lines

File: src/components/VDatePicker/VDatePicker.vue [hunk 6]
```diff
@@ -72,6 +73,7 @@
     },
 
     computed: {
+      supportsLocaleFormat: () => supportsLocaleFormat,
```
Comment (KaelWD) [inline src/components/VDatePicker/VDatePicker.vue]: Does this need to be reactive? I don't think the browser's javascript support is likely to change at runtime

**Comment by jacekkarczmarczyk** [comment]
PR updated

**Pull request merged**